python/browser_widget/worker.py

```diff
@@ -53,7 +66,7 @@ def stop(self, wait_for_completion=True):
             self._wait_condition.notifyAll()
```

Comment:
I have a feeling we should also call `self._notifier.deleteLater` before leaving this method to ensure the qt object is deleted.